tools/toolchains/mbed_toolchain.py

```diff
@@ -957,6 +957,12 @@ def add_linker_defines(self):
                 self.ld.append(define_string)
                 self.flags["ld"].append(define_string)
 
+        if "XIP_ENABLE" in self.target.macros :
```

Comment:
Please comment on what XIP is, and why it needs to be added here.

Reply:
XIP stands for Execute In Place. Since the text/RO data gets too big to fit on on chip flash, we need to move some code to the QSPI to execute for bigger applications such as Pelion cloud client. This feature is designed to be enabled in mbed_app.json because each application usually has its own JSON. By default it's disabled, because if the text/RO data can fit on the internal flash, there's no need to enable this feature.